lib/linkify.dart

```diff
@@ -152,17 +192,30 @@ List<LinkifyElement> linkify(
           list.add(LinkElement(urlMatch.group(2)));
         }
       }
-    } else if (emailMatch != null) {
-      text = text.replaceFirst(emailMatch.group(0), "");
+    } else if (formattedEmailMatch != null) {
+      text = text.replaceFirst(formattedEmailMatch.group(0), "");
 
-      if (emailMatch.group(1).isNotEmpty) {
-        list.add(TextElement(emailMatch.group(1)));
+      if (formattedEmailMatch.group(3).isNotEmpty) {
+        list.add(TextElement(formattedEmailMatch.group(3)));
       }
 
-      if (emailMatch.group(2).isNotEmpty) {
+      if (formattedEmailMatch.group(4).isNotEmpty) {
         // Always humanize emails
         list.add(EmailElement(
-          emailMatch.group(2).replaceFirst(RegExp(r"mailto:"), ""),
+          formattedEmailMatch.group(4).replaceFirst(RegExp(r"mailto:"), ""),
+        ));
+      }
+    } else if (looseEmailMatch != null && formattedEmailMatch == null) {
```

Comment:
testing formattedEmailMatch == null is redundant here.

Reply:
Is possible to match looseEmail and formattedEmail in the same expression. eg: "[e-mail](mailto:person@example.com)" this will be matched in both cases and I don't think we should evaluate two times.